### ISSUE TITLE
iOS second resolve causes app to crash

### ISSUE THREAD

**Issue by DeliriumCode** [dadhi/DryIoc#285]
So I am getting the following error on the latest version of xamarin forms on iOS. It happens on the second resolve of MyClass, the first time it works absolutely fine, but the second time resolve is called the app crashes. There is an example below of myclass and how i am initializing the ioc container and using it to resolve.  Please note, that this is a transient ioc instance, and i am using resolve to create a new instance, with dependency injection.

On a side note, thank you so much for creating and maintaining this project. Its amazing!!!

Stack Trace:
```
Error: System.PlatformNotSupportedException: 'Operation is not supported on this platform.'

	0x9D in FastExpressionCompiler.LightExpression.ExpressionCompiler.TryCompileBoundToFirstClosureParam at \packages\dryioc\4.2.1\contentFiles\cs\netstandard2.0\DryIoc\FastExpressionCompiler.cs:328,13	C#
 	0x76 in DryIoc.FactoryDelegateCompiler.CompileToFactoryDelegate at \packages\dryioc\4.2.1\contentFiles\cs\netstandard2.0\DryIoc\Container.cs:3892,17	C#
 	0x104 in DryIoc.Container.DryIoc.IResolver.Resolve at \packages\dryioc\4.2.1\contentFiles\cs\netstandard2.0\DryIoc\Container.cs:338,25	C#
 	0xFFFFFFFFFFFFFFFF in DryIoc.Resolver.Resolve<MyApp.ViewModel> at \packages\dryioc\4.2.1\contentFiles\cs\netstandard2.0\DryIoc\Container.cs	C#
```

//MyClass
```
public class MyClass
{
public MyClass(ISettingsService settingsService   )
{
}
}
public interface ISettingsService {}
```

Error: System.PlatformNotSupportedException: 'Operation is not supported on this platform.'
//Initialized and consumed as:
```
public class IOC {
public static DryIoc.Container Container { get; set; } = new DryIoc.Container();
public static void Initialise()
{
	Container.Register<ViewModel>();
}
public static MyClass Resolve()
{
return Container.Resolve<MyClass>();
}
}
```

//Consumed as:

**Comment by dadhi** [comment]
Hi @DeliriumCode ,

You need to add this rule into your container for iOS `container = container.With(rules => rules.WithUseInterpretation())`.
See the #156 for details - scroll to the bottom.

**Comment by DeliriumCode** [comment]
Thanks dadhi,
Yep that fixed the issue perfectly! Thanks for the quick response on this :)

For anyone else who sees this issue, I am now using:

 ```
            if (Xamarin.Forms.Device.RuntimePlatform == Xamarin.Forms.Device.iOS)
                Container = new DryIoc.Container(rules: Rules.Default.WithUseInterpretation());
            else
                Container = new DryIoc.Container();
```

**Comment by awalker-dsg** [comment]
This is an old issue, but I'm adding a comment because we recently ran into a similar issue in a mobile app ported from Xamarin to .NET MAUI that targets net8.0.

Our app was crashing only in Release builds of iOS; Debug builds did not have any problems. We also run on Android, and had no issues there.

The crash only occurred after a call to `PopToRootAsync` was followed by a call to `PushAsync` using a page that had previously been resolved. Also, we saw a different exception from the one reported in the OP:

    System.NullReferenceException: Arg_NullReferenceException
    at Microsoft.Maui.Controls.Internals.NavigationProxy.PushAsync(Page root, Boolean animated)
    at Microsoft.Maui.Controls.Internals.NavigationProxy.PushAsync(Page root)

[This change](https://github.com/dadhi/DryIoc/issues/285#issuecomment-641641199) suggested by @DeliriumCode via @dadhi fixed the issue.
